code/modules/mob/living/carbon/human/examine.dm

```diff
@@ -56,17 +56,22 @@
 
 	var/list/nospecies = list("Abductor", "Shadowling", "Neara", "Monkey", "Stok", "Farwa", "Wolpin") //species that won't show their race no matter what
 
-	if (skipjumpsuit && skipface || (species.name in nospecies)) //either obscured or on the nospecies list
-		msg += "!\n"	//omit the species when examining
-	else if (species.name == "Slime People") //snowflakey because Slime People are defined as a plural
+	var/displayed_species = get_species()
+	for(var/obj/item/clothing/C in src)			//Disguise checks
+		if (C == src.head || C == src.wear_suit || C == src.wear_mask || C == src.w_uniform || C == src.belt || C == src.back)

```

Comment:
This is very ugly, but for some odd reason there doesn't actually seem to be a clean way to do this.